src/main/java/com/xmlcalabash/library/HttpRequest.java

```diff
@@ -561,28 +563,25 @@ private void doPutOrPostSinglepart(HttpEntityEnclosingRequest method, XdmNode bo
         if (encoding != null && !"base64".equals(encoding)) {
             throw XProcException.stepError(52);
         }
+        
+        HttpEntity requestEntity = null;
 
         try {
             if ("base64".equals(encoding)) {
-                String charset = body.getAttributeValue(_charset);
```

Comment:
The base64-encoded data would typically be non-textual in nature, but even if it is in fact text which has been base64-encoded for some reason, and therefore does have a charset, this charset would be irrelevant to the task at hand which is just to upload the raw byte stream.